### PR TITLE
UCP and allowed media hosts

### PR DESCRIPTION
**Pull request by modelrailroader**
I moved the ucp to twig while I build the button from #3010 

And I added a new config item where users can add media hosts for videos or pictures manually. I think, this really makes sense because otherwise it only can be changed in the code.

### CODE REVIEW

File: phpmyfaq/assets/src/user/ucp.js [hunk 1]
```diff
@@ -13,8 +13,9 @@
  * @since     2024-03-03
  */
 
-import { updateUserControlPanelData } from '../api';
+import {removeTwofactorConfig, updateUserControlPanelData} from '../api';
```
Comment (thorsten) [inline phpmyfaq/assets/src/user/ucp.js:16]: Please add spaces around the imported function names for better readability

File: phpmyfaq/assets/src/user/ucp.js [hunk 2]
```diff
@@ -13,8 +13,9 @@
...
 import { addElement } from '../utils';
+import {pushErrorNotification, pushNotification} from "../../../admin/assets/src/utils";
```
Comment (thorsten) [inline phpmyfaq/assets/src/user/ucp.js:18]: Please add spaces around the imported function names for better readability

File: phpmyfaq/assets/src/user/ucp.js [hunk 3]
```diff
@@ -47,5 +48,18 @@ export const handleUserControlPanel = () => {
         );
       }
     });
+
+    const confirmRemoveTwofactor = document.getElementById('pmf-remove-twofactor-confirm');
+    confirmRemoveTwofactor.addEventListener('click', async (event) => {
```
Comment (thorsten) [inline phpmyfaq/assets/src/user/ucp.js:53]: A check is missing if "confirmRemoveTwofactor" is already initiated.

File: phpmyfaq/assets/src/user/ucp.js [hunk 4]
```diff
@@ -47,5 +48,18 @@ export const handleUserControlPanel = () => {
...
+      event.preventDefault();
+      const csrfToken = document.getElementById('pmf-csrf-token-remove-twofactor');
+      const response = await removeTwofactorConfig(csrfToken.value);
+      if (response.success) {
+        window.location.reload();
```
Comment (thorsten) [inline phpmyfaq/assets/src/user/ucp.js:58]: Instead of reloading, why not change the UI?

File: phpmyfaq/assets/templates/ucp.twig
Comment (thorsten) [inline phpmyfaq/assets/templates/ucp.twig:1]: WIth the Twig template, you can delete the old one

File: phpmyfaq/src/phpMyFAQ/Controller/Frontend/UserController.php [hunk 5]
```diff
@@ -19,12 +19,14 @@
 
 use phpMyFAQ\Controller\AbstractController;
 use phpMyFAQ\Core\Exception;
+use phpMyFAQ\Export\Json;
```
Comment (thorsten) [inline phpmyfaq/src/phpMyFAQ/Controller/Frontend/UserController.php:22]: Why do we need the Json Export class here?

File: phpmyfaq/src/phpMyFAQ/Controller/Frontend/UserController.php [hunk 6]
```diff
@@ -251,4 +250,31 @@ public function requestUserRemoval(Request $request): JsonResponse
...
         }
     }
+
+    /**
+     * @throws Exception
+     */
+    #[Route('api/user/remove-twofactor', methods: ['POST'])]
+    public function removeTwofactorConfig(Request $request): JsonResponse
+    {
+        $data = json_decode($request->getContent());
```
Comment (thorsten) [inline phpmyfaq/src/phpMyFAQ/Controller/Frontend/UserController.php:260]: Please add a check if the user is registered.

File: phpmyfaq/src/phpMyFAQ/Setup/Installer.php [hunk 7]
```diff
@@ -355,6 +355,7 @@ class Installer extends Setup
         'records.numberMaxStoredRevisions' => '10',
         'records.enableAutoRevisions' => 'false',
         'records.orderStickyFaqsCustom' => 'false',
+        'records.allowedMediaHosts' => 'www.youtube.com',
```
Comment (thorsten) [inline phpmyfaq/src/phpMyFAQ/Setup/Installer.php:358]: This has to be added in the Update class as well.

File: phpmyfaq/src/phpMyFAQ/Helper/FaqHelper.php [hunk 8]
```diff
@@ -220,6 +220,8 @@ public function createFaqUrl(FaqEntity $faqEntity, int $categoryId): string
     public function cleanUpContent(string $content): string
     {
         $contentLength = strlen($content);
+        $allowedMediaHosts = explode(',', $this->configuration->get('records.allowedMediaHosts'));
```
Comment (thorsten) [inline phpmyfaq/src/phpMyFAQ/Helper/FaqHelper.php:223]: This should be moved to an own method in the Configuration class.

File: phpmyfaq/translations/language_en.php [hunk 9]
```diff
@@ -1459,4 +1459,9 @@
 $PMF_LANG['msgTrendingFAQs'] = 'Trending FAQs';
 $PMF_LANG['msgGenerateApiClientToken'] = 'Generate API Client Token';
 
+// added v4.0.0-alpha.3 - 2024-06-19 by Jan
+$LANG_CONF['records.allowedMediaHosts'] = ['area', 'Allowed external hosts for media content (seperate with commas)'];
```
Comment (thorsten) [inline phpmyfaq/translations/language_en.php:1463]: Typo: separate

**Review by thorsten — CHANGES_REQUESTED** [review]

File: phpmyfaq/src/phpMyFAQ/Controller/Frontend/UserController.php (hunk 6 above)
Comment (modelrailroader) [reply phpmyfaq/src/phpMyFAQ/Controller/Frontend/UserController.php:260]: Not necessary at this time, because we only use CurrentUser.

**Comment by modelrailroader** [comment]
All requested changed are implemented.

**Comment by modelrailroader** [comment]
I resolved the merge conflict.

**Pull request merged**